instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/instrumentation/annotations/InstrumentationAnnotationsAutoConfiguration.java

```diff
@@ -8,20 +8,35 @@
 import io.opentelemetry.api.OpenTelemetry;
 import io.opentelemetry.instrumentation.annotations.WithSpan;
 import org.aspectj.lang.annotation.Aspect;
+import org.springframework.boot.autoconfigure.condition.AllNestedConditions;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.core.DefaultParameterNameDiscoverer;
 import org.springframework.core.ParameterNameDiscoverer;
 
 /** Configures {@link WithSpanAspect} to trace bean methods annotated with {@link WithSpan}. */
 @ConditionalOnBean(OpenTelemetry.class)
 @ConditionalOnClass(Aspect.class)
-@ConditionalOnProperty(name = "otel.instrumentation.annotations.enabled", matchIfMissing = true)
+@Conditional(InstrumentationAnnotationsAutoConfiguration.Condition.class)
 @Configuration
 public class InstrumentationAnnotationsAutoConfiguration {
+
+  static final class Condition extends AllNestedConditions {
```

Comment:
Just curious why you implemented it this way. Would having
```
@ConditionalOnProperty(name = "otel.instrumentation.annotations.enabled", matchIfMissing = true)
@Conditional(SdkEnabled.class)
```
where `SdkEnabled` is shared by all auto configuration classes also work?

Reply:
Interesting idea! I hadn't considered that - I'll try.